vm.go

```diff
@@ -102,13 +102,19 @@ func (l *State) equalObjects(t1, t2 value) bool {
 	case *userData:
 		if t1 == t2 {
 			return true
+		} else if t2, ok := t2.(*userData); ok && t2 != nil {
+			tm = l.equalTagMethod(t1.metaTable, t2.metaTable, tmEq)
+		} else if !ok || t2 == nil {
+			return false
```

Comment:
We make some effort to ensure that `nil` values are `(nil, nil)`, so if we panic because we get a `(*userData, nil)` here (or `(*table, nil)` below) then we've done something wrong somewhere else. Given that, it's sufficient to:
```go
} else if t2, ok := t2.(*userData); ok {
	tm = l.equalTagMethod(t1.metaTable, t2.metaTable, tmEq)
}
return false
```